deeplearning4j-core/src/main/java/org/deeplearning4j/nn/layers/BaseLayer.java

```diff
@@ -300,7 +314,7 @@ public INDArray preOutput(INDArray x, boolean training) {
         INDArray b = getParam(DefaultParamInitializer.BIAS_KEY);
         INDArray W = getParam(DefaultParamInitializer.WEIGHT_KEY);
         if(conf.isUseDropConnect() && training) {
-            if (conf.getLayer().getDropOut() > 0) {
+            if (conf.getLayer().getDropOut() > 0 && training) {
```

Comment:
`training` is already checked in line 302, I think it's unnecessary to put it here also?